checks.d/linux_proc_extras.py

```diff
@@ -24,47 +25,66 @@
 }
 
 class MoreUnixCheck(AgentCheck):
+    tags = []
+    proc_path_map = {
+        "inode_info": "/proc/sys/fs/inode-nr",
+        "stat_info": "/proc/stat",
+        "entropy_info": "/proc/sys/kernel/random/entropy_avail",
+    }
+
     def check(self, instance):
-        tags = instance.get('tags', [])
+        self.tags = instance.get('tags', [])
+        self.set_paths()
 
-        state_counts = defaultdict(int)
 
-        prio_counts = defaultdict(int)
+        self.get_inode_info()
+        self.get_stat_info()
+        self.get_entropy_info()
+        self.get_process_states()
 
-        proc_location = self.agentConfig.get('procfs_path', '/proc').rstrip('/')
+    def set_paths(self):
+        proc_location = self.agentConfig.get('procfs_path', '/proc').rstrip('/').lstrip('/')
 
-        proc_path_map = {
+        self.proc_path_map = {
             "inode_info": "sys/fs/inode-nr",
             "stat_info": "stat",
             "entropy_info": "sys/kernel/random/entropy_avail",
         }
 
-        for key, path in proc_path_map.iteritems():
-            proc_path_map[key] = "{procfs}/{path}".format(procfs=proc_location, path=path)
+        for key, path in self.proc_path_map.iteritems():
+            self.proc_path_map[key] = "{procfs}/{path}".format(procfs=proc_location, path=path)
+
 
-        with open(proc_path_map['inode_info'], 'r') as inode_info:
+    def get_inode_info(self):
+        with open(self.proc_path_map['inode_info'], 'r') as inode_info:
             inode_stats = inode_info.readline().split()
-            self.gauge('system.inodes.total', float(inode_stats[0]), tags=tags)
-            self.gauge('system.inodes.used', float(inode_stats[1]), tags=tags)
+            self.gauge('system.inodes.total', float(inode_stats[0]), tags=self.tags)
+            self.gauge('system.inodes.used', float(inode_stats[1]), tags=self.tags)
 
-        with open(proc_path_map['stat_info'], 'r') as stat_info:
+    def get_stat_info(self):
+        with open(self.proc_path_map['stat_info'], 'r') as stat_info:
             lines = [line.strip() for line in stat_info.readlines()]
-
             for line in lines:
                 if line.startswith('ctxt'):
                     ctxt_count = float(line.split(' ')[1])
-                    self.monotonic_count('system.linux.context_switches', ctxt_count, tags=tags)
+                    self.monotonic_count('system.linux.context_switches', ctxt_count, tags=self.tags)
                 elif line.startswith('processes'):
+                    self.log.info(line)
```

Comment:
Do we really want this at INFO level? Side note: if we're only reading the log, without knowing the check, this line doesn't make sense. Could we prefix it with what it is/means?